lib/index.d.ts

```diff
@@ -0,0 +1,177 @@
+
+import {
+    IncomingHttpHeaders,
+    IncomingMessage,
+    OutgoingHttpHeaders,
+    ServerResponse
+} from 'http';
+import { Readable, Stream } from 'stream';
+import { UrlObject } from 'url';
+
+
+export interface InjectedRequest extends Readonly<Readable> {
+    readonly httpVersion: '1.1';
+    readonly method: string;
+    readonly url: string;
+    readonly headers: Readonly<IncomingHttpHeaders>;
+    readonly connection: {
+        readonly remoteAddress: string;
+    };
+}
+
+export type MaybeInjectedRequest = InjectedRequest | IncomingMessage;
+
+export interface ResponseObject {
+    /**
+     * An object containing the raw request and response objects.
+     */
+    raw: {
+        /**
+         * The simulated request object.
+         */
+        req: InjectedRequest;
+
+        /**
+         * The simulated response object.
+         */
+        res: ServerResponse;
+    };
+
+    /**
+     * An object containing the response headers.
+     */
+    headers: OutgoingHttpHeaders;
+
+    /**
+     * The HTTP status code.
+     */
+    statusCode: number;
+
+    /**
+     * The HTTP status message.
+     */
+    statusMessage: string;
+
+    /**
+     * The payload as a UTF-8 encoded string.
+     */
+    payload: string;
+
+    /**
+     * The raw payload as a Buffer.
+     */
+    rawPayload: Buffer;
+
+    /**
+     * An object containing the response trailers
+     */
+    trailers: NodeJS.Dict<string>;
```

Comment:
Does TypeScript provide a global `NodeJS` module with the Dict type? First time I'm seeing this. I couldn't find documentation on it.

Reply:
I did it like that for compatibility with https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9632b0d2ffbea7376d516f462aa53b327cd6b03f/types/node/http.d.ts#L331.

But we can just use [the type](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9632b0d2ffbea7376d516f462aa53b327cd6b03f/types/node/globals.d.ts#L607-L609) directly, if you think it will be better? Or maybe set it to `IncomingMessage['trailers']`?

Reply:
Actually, I just realised that it is our own property, with its own semantics, so it should probably just be typed directly, eg.:

```ts
trailers: { [header: string]: string }
```

Reply:
Should we apply this as-is?
```suggestion
    trailers: { [header: string]: string };
```